### ISSUE TITLE
Can't map data to `alpha` values

### ISSUE THREAD

**Issue by ssfrr** [MakieOrg/AlgebraOfGraphics.jl#482]
## Bug description

I'm trying to map data to the `alpha` keyword, to create varying transparency for different data points. When I try to do so, it throws a Method error.

## Steps to reproduce

This code reproduces the issue:

```julia
draw(
    data((;
        x=rand(100),
        y=rand(100),
        z=rand(100),
        animal=rand(["cat", "dog", "horse"], 100))) *
    mapping(:x, :y, color=:animal, alpha=:z) *
    visual(Scatter))
```

It throws an error:

```
ERROR: MethodError: no method matching to_color(::Tuple{ColorTypes.RGBA{Float32}, Vector{Float64}})

Closest candidates are:
  to_color(::Scene, ::Any, ::Cycled)
   @ Makie ~/.julia/packages/Makie/6NLuU/src/makielayout/blocks/axis.jl:691
  to_color(::Makie.Palette)
   @ Makie ~/.julia/packages/Makie/6NLuU/src/conversions.jl:845
  to_color(::Symbol)
   @ Makie ~/.julia/packages/Makie/6NLuU/src/conversions.jl:854
  ...

Stacktrace:
  [1] (::Makie.var"#111#113")(color::Tuple{ColorTypes.RGBA{Float32}, Vector{Float64}}, a::Float64)
    @ Makie ~/.julia/packages/Makie/6NLuU/src/colorsampler.jl:0
  [2] map(f::Makie.var"#111#113", scene::Union{…}, arg1::Observable{…}, args::Observable{…}; ignore_equal_values::Bool, priority::Int64)
    @ Makie ~/.julia/packages/Makie/6NLuU/src/scenes.jl:160
  [3] map
    @ ~/.julia/packages/Makie/6NLuU/src/scenes.jl:157 [inlined]
  [4] assemble_colors(colortype::Tuple{ColorTypes.RGBA{…}, Vector{…}}, color::Observable{Any}, plot::Scatter{Tuple{…}})
    @ Makie ~/.julia/packages/Makie/6NLuU/src/colorsampler.jl:354
  [5] color_and_colormap!(plot::Scatter{Tuple{Vector{Point{2, Float32}}}}, colors::Observable{Any})
    @ Makie ~/.julia/packages/Makie/6NLuU/src/interfaces.jl:14
  [6] color_and_colormap!(plot::Scatter{Tuple{Vector{Point{2, Float32}}}})
    @ Makie ~/.julia/packages/Makie/6NLuU/src/interfaces.jl:10
  [7] calculated_attributes!
    @ Makie ~/.julia/packages/Makie/6NLuU/src/interfaces.jl:64 [inlined]
  [8] connect_plot!(parent::Scene, plot::Scatter{Tuple{Vector{Point{2, Float32}}}})
    @ Makie ~/.julia/packages/Makie/6NLuU/src/interfaces.jl:253
...
```

## Version info
```
julia> versioninfo()
Julia Version 1.10.0-rc1
Commit 5aaa9485436 (2023-11-03 07:44 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 8 × Apple M2
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 1 on 4 virtual cores
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 

(src) pkg> status
Status `~/Library/CloudStorage/Dropbox/Documents/Heaviside Research/Clients/Figure8/MAPFRE Initial Analysis/src/Project.toml`
  [cbdf2221] AlgebraOfGraphics v0.6.17
  [e28b5b4c] Bootstrap v2.4.0
  [336ed68f] CSV v0.10.11
  [13f3f980] CairoMakie v0.11.3
  [8be319e6] Chain v0.5.0
  [a93c6f00] DataFrames v1.6.1
  [6f286f6a] MultivariateStats v0.10.2
⌅ [2913bbd2] StatsBase v0.33.21
  [24678dba] TSne v1.3.0
  [c4f8c510] UMAP v0.1.10
  [276b4fcb] WGLMakie v0.9.2
```

**Comment by ssfrr** [comment]
Also, an additional note - I spent a bit of time trying to figure out how to manually create a vector of color values that would include my custom alpha, but ran into a couple issues

1. I wasn't able to figure out how to duplicate the color group cycling logic
2. I realized that even if I got it to work, it would probably mess up my data grouping and legend, because the `color` attribute is used to determine the color groups

Basically what I'd be looking for would be that the `color` field would still get my `Vector{String}`, which would set the base color and also the grouping (and the labels on the legend), and then the `alpha` attribute would basically be added as an additional attribute on top (maybe by multiplying the alpha channel of the existing values from the colormap?)

**Comment by japhir** [comment]
Also the much simpler (only one alpha value per line) runs into this issue:

```julia
using GLMakie
using AlgebraOfGraphics
using DataFrames

dat = DataFrame(
    x = 1:10,
    y = 11:20,
    z = ['a','a','a','b','b','b','c','c','c','c']
)

data(dat) *
    mapping(:x, :y,
            # color=:z, # works fine
            alpha=:z, # errors
            group=:z) *
    visual(Lines) |> draw
```

<details>
<summary> unfold for full error </summary>

ERROR: ArgumentError: ProcessedLayer with plot type Lines did not have :alpha in its AestheticMapping. The mapping was {1 = AlgebraOfGraphics.AesX, 2 = AlgebraOfGraphics.AesY, :color = AlgebraOfGraphics.AesColor, :linestyle = AlgebraOfGraphics.AesLineStyle}
Stacktrace:
  [1] hardcoded_or_mapped_aes(processedlayer::ProcessedLayer, key::Symbol, aes_mapping::Dictionaries.Dictionary{Union{Int64, Symbol}, Type{<:AlgebraOfGraphics.Aesthetic}})
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:243
  [2] (::AlgebraOfGraphics.var"#p!#134"{Set{Type{<:AlgebraOfGraphics.Aesthetic}}, Dictionaries.Dictionary{Union{Int64, Symbol}, Type{<:AlgebraOfGraphics.Aesthetic}}, ProcessedLayer, Dictionaries.Dictionary{Union{Int64, Symbol}, Type{<:AlgebraOfGraphics.Aesthetic}}})(key::Symbol)
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:167
  [3] foreach(f::AlgebraOfGraphics.var"#p!#134"{Set{Type{<:AlgebraOfGraphics.Aesthetic}}, Dictionaries.Dictionary{Union{Int64, Symbol}, Type{<:AlgebraOfGraphics.Aesthetic}}, ProcessedLayer, Dictionaries.Dictionary{Union{Int64, Symbol}, Type{<:AlgebraOfGraphics.Aesthetic}}}, d::Dictionaries.Indices{Symbol})
    @ Dictionaries ~/.julia/packages/Dictionaries/RaTRN/src/foreach.jl:36
  [4] compute_scale_properties(processedlayers::Vector{ProcessedLayer}, scales::AlgebraOfGraphics.Scales)
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:172
  [5] compute_axes_grid(d::Layer, scales::AlgebraOfGraphics.Scales; axis::Dictionaries.Dictionary{Symbol, Any})
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:252
  [6] compute_axes_grid
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:248 [inlined]
  [7] compute_axes_grid(fig::Figure, d::Layer, scales::AlgebraOfGraphics.Scales; axis::Dictionaries.Dictionary{Symbol, Any})
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:218
  [8] compute_axes_grid
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/algebra/layers.jl:215 [inlined]
  [9] #363
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:20 [inlined]
 [10] update
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:10 [inlined]
 [11] plot!(fig::Figure, d::Layer, scales::AlgebraOfGraphics.Scales; axis::Dictionaries.Dictionary{Symbol, Any})
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:20
 [12] plot!
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:16 [inlined]
 [13] (::AlgebraOfGraphics.var"#374#379"{Dictionaries.Dictionary{Symbol, Any}, Dictionaries.Dictionary{Symbol, Any}, Dictionaries.Dictionary{Symbol, Any}, Dictionaries.Dictionary{Symbol, Any}, Layer, AlgebraOfGraphics.Scales, AlgebraOfGraphics.FigureSettings})(f::Figure)
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:162
 [14] update
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:10 [inlined]
 [15] _draw(d::Layer, scales::AlgebraOfGraphics.Scales; axis::Dictionaries.Dictionary{Symbol, Any}, figure::Dictionaries.Dictionary{Symbol, Any}, facet::Dictionaries.Dictionary{Symbol, Any}, legend::Dictionaries.Dictionary{Symbol, Any}, colorbar::Dictionaries.Dictionary{Symbol, Any})
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:161
 [16] _draw
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:154 [inlined]
 [17] draw(d::Layer, scales::AlgebraOfGraphics.Scales; axis::@NamedTuple{}, figure::@NamedTuple{}, facet::@NamedTuple{}, legend::@NamedTuple{}, colorbar::@NamedTuple{}, palette::Nothing)
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:85
 [18] draw
    @ ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:75 [inlined]
 [19] draw(d::Layer)
    @ AlgebraOfGraphics ~/.julia/packages/AlgebraOfGraphics/BPWA1/src/draw.jl:75
 [20] |>(x::Layer, f::typeof(draw))
    @ Base ./operators.jl:917
 [21] top-level scope
    @ none:16


</details>
